### ISSUE TITLE
C: --_xformat ignoring the struct key word while extracting function information the header

### ISSUE THREAD

**Issue by sourabhjains** [universal-ctags/ctags#1772]
I am extracting function information in the following format.
**function_name:return_type:signature**

I have used following command to do that
**ctags --language-force=C -x --_xformat="%N:%{typeref}:%{signature}"  --kinds-C=p foo.h**

When I formatting the ouput using --_xformat with typeref tag. typeref doesn't consider struct keyword while formatting the ouput.

for example: foo.h
#include <stdio.h>
int first (int a, int b);
struct mystructure second (int a);
long long third (long a);

The ouput of above command will be:
first:int:(int a,int b)
second:mystructure:(int a)
third:long long:(long a)

Here, in second function --_xformat misssed the struct keyword.

**Comment by masatake** [comment]
Thank you for reporting.
I will work on this item.

**Comment by masatake** [comment]
See #1773.
With the change proposed in #1773, the command line makes following output:
```
first:typename:int:(int a,int b)
second:struct:mystructure:(int a)
third:typename:long long:(long a)
```

This output is not enough for your requirement. However, I don't think I should make ctags specialized to your requirement more. sed may help you as usual.

It is possible to introduce typerefPrefix and typerefName fields. See the commit log of the change if you are interested in introducing new fields.

**Comment by masatake** [comment]
I merged #1773. The change may solve your issue partially. As I wrote you need a post processor like sed for your purpose.
